mailbox/api/src/test/java/org/apache/james/mailbox/manager/IntegrationResources.java

```diff
@@ -21,28 +21,28 @@
 
 import org.apache.james.mailbox.MailboxManager;
 import org.apache.james.mailbox.MessageIdManager;
-import org.apache.james.mailbox.quota.MaxQuotaManager;
-import org.apache.james.mailbox.quota.QuotaManager;
 import org.apache.james.mailbox.acl.GroupMembershipResolver;
 import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.quota.MaxQuotaManager;
+import org.apache.james.mailbox.quota.QuotaManager;
 import org.apache.james.mailbox.quota.QuotaRootResolver;
 
 /**
  * Provides empty resources for integration tests.
  */
-public interface IntegrationResources {
+public interface IntegrationResources<T extends MailboxManager> {
```

Comment:
how many warning introduced ? :trollface: 
Can you explain the goal of being able to retrieve the exact MailboxManager type ?

Reply:
I explained in the daily and in the previous pr.

This class is stateless so messageIdManager needs to be generated from mailbox manager. What is impossible without the specific type. Raphaël and I discussed and looked for other solutions.



About the warning, Raphael solve them all in eclipse, and I did not introduced additional ones in intelliJ.

Reply:
For your initial question, the answer is simple: because of separation between api and store...

Reply:
I don't buy it but there's three 👍 so I will merge